nsc/policy/forms.py

```diff
@@ -128,3 +132,76 @@ def __init__(self, *args, **kwargs):
 
         self.fields["policy"].widget.attrs.update({"class": "govuk-select"})
         self.fields["policy"].queryset = Policy.objects.all()
+
+
+class ArchiveForm(forms.ModelForm):
+    archived_reason_error = _(
+        "You must add a public statement before pressing Archive."
+    )
+    archived_reason_initial = _("This condition has been archived because:")
+
+    archived_reason = forms.CharField(
+        required=True,
+        label=_("Why has this form recommendation been archived?"),
+        help_text=_(
+            "Write a public statement to explain reasons for archiving recommendation (mandatory)"
+            "<br/>"
+            "Use markdown to format the text."
+        ),
+        widget=forms.Textarea,
+        initial=archived_reason_initial,
+    )
+
+    class Meta:
+        model = Policy
+        fields = ["archived_reason"]
+
+    def clean_archived_reason(self):
+        value = self.cleaned_data["archived_reason"]
+
+        if not value:
+            raise ValidationError(self.archived_reason_error)
+
+        if value == self.archived_reason_initial:
+            raise ValidationError(self.archived_reason_error)
```

Comment:
Not sure why this check is here, I'm guessing its to make sure the message has been updated. In that case the message should be updated otherwise it will be confusing when people are told to enter text but there is text present.

Reply:
Yeah, it's just to check that they haven't just left the default text in place - it looked in the design that they wanted to have a template. Can change the message.